Mathlib/Algebra/MvPolynomial/NoZeroDivisors.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2025 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+module
+
+public import Mathlib.Algebra.MvPolynomial.Variables
+public import Mathlib.Algebra.MvPolynomial.Equiv
+
+/-!
+# Multivariate polynomials over `NoZeroDivisors`
+
+Many results about polynomials hold when the coefficient ring has no zero divisors.
+
+This file does not define any new operations, but proves some of these stronger results.
+
+## TODOs
+
+* Add a `totalDegree_mul_eq` theorem, which states that the total degree of a product of two
+nonzero multivariate polynomials is the sum of their total degrees. (See also
+`MvPolynomial.totalDegree_mul_of_isDomain`, which proves this
+under the assumption that the coefficient ring has cancellative multiplication.)
```

Comment:
I am slightly confused. This is definitely not true if the ring has zero divisors?